convert/what3words/src/index.ts

```diff
@@ -0,0 +1,85 @@
+import { FlatfileClient } from '@flatfile/api'
+import type { FlatfileEvent } from '@flatfile/listener'
+import {
+  bulkRecordHook,
+  type FlatfileRecord,
+} from '@flatfile/plugin-record-hook'
+import what3words, {
+  ApiVersion,
+  Transport,
+  What3wordsService,
+  fetchTransport,
+} from '@what3words/api'
+
+const api = new FlatfileClient()
+
+interface What3WordsConfig {
+  what3wordsField: string
+  countryField: string
+  nearestPlaceField: string
+  latField: string
+  longField: string
+  sheetSlug?: string
+}
+
+async function getSecret(
+  name: string,
+  environmentId?: string,
+  spaceId?: string
+): Promise<string | undefined> {
+  try {
+    const secrets = await api.secrets.list({ spaceId, environmentId })
+    const secret = secrets.data.find((secret) => secret.name === name)
+    if (!secret) {
+      return undefined
+    }
+    return secret.value
+  } catch (e) {
+    console.error(e, `Error fetching secret ${name}`)
+    return undefined
+  }
+}
+
+export function convertWhat3words(config: What3WordsConfig) {
+  const w3wConfig: {
+    host: string
+    apiVersion: ApiVersion
+  } = {
+    host: 'https://api.what3words.com',
+    apiVersion: ApiVersion.Version3,
+  }
+  return bulkRecordHook(
+    config.sheetSlug || '**',
+    async (records: FlatfileRecord[], event: FlatfileEvent) => {
+      const { environmentId, spaceId } = event.context
+
+      const apiKey = await getSecret('W3W_API_KEY', environmentId, spaceId)
```

Comment:
We should use `event.secrets()` here
```suggestion
      const apiKey = await event.secrets('W3W_API_KEY')
```